core/src/tasks/recordProperty/importRecordProperty.ts

```diff
@@ -70,7 +72,33 @@ export class ImportRecordProperty extends RetryableTask {
       hash[property.id] = Array.isArray(propertyValues)
         ? propertyValues
         : [propertyValues];
-      await record.addOrUpdateProperties(hash);
+
+      try {
+        await record.addOrUpdateProperties(hash);
+      } catch (error) {
+        if (property.unique && error instanceof UniqueConstraintError) {
+          await CLS.afterCommit(() =>
```

Comment:
This is a little weird but without doing a locking read, we have to "spoil" the transaction to know that there was a write error to get a UniqueConstraintError